### PR TITLE
Doc: fixed missing behavior explaination of sort=false for groupby issue #47529

### LINKED ISSUE #47529: DOC: missed behavior explaination of sort=False for groupby
### Pandas version checks

- [x] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.groupby.html

### Documentation problem

The docs miss an explanation of sort=False for groupby. Does the order among groups with their keys follow the order of appearance of the keys in the original data frame? Or the groups may be out of order?

### Suggested fix for documentation

When setting sort=False for groupby. One may want the order among groups with their keys follows the order of appearance of the keys in the original data frame. Can this be guaranteed?

### PR DESCRIPTION
**Pull request by raanasn**
- [x] closes #47529 
- [x] All code checks passed

Changes were made by @hubschne and @raanasn at the python conference 2023 Berlin workshop

### CODE REVIEW

**Comment by raanasn** [comment]
Reviewer suggestions: @phofl, @noatamir, @jorisvandenbossche and @marcogorelli

File: pandas/core/shared_docs.py [hunk 1]
```diff
@@ -130,7 +130,9 @@
 sort : bool, default True
     Sort group keys. Get better performance by turning this off.
     Note this does not influence the order of observations within each
-    group. Groupby preserves the order of rows within each group.
+    group. Groupby preserves the order of rows within each group. If False
+    the order among groups with their keys follows the order of appearance
+    of the keys in the original dataframe.
```
Comment (MarcoGorelli) [inline pandas/core/shared_docs.py:135]:
```suggestion
    group. Groupby preserves the order of rows within each group. If False,
    the groups will appear in the same order as they did in the original DataFrame.
```

File: doc/source/user_guide/groupby.rst [hunk 2]
```diff
@@ -205,7 +205,7 @@ By default the group keys are sorted during the ``groupby`` operation. You may h
    df2.groupby(["X"], sort=False).sum()
 
 
-Note that ``groupby`` will preserve the order in which *observations* are sorted *within* each group.
+Note that ``groupby`` will preserve the order in which *observations* are sorted *within* each group .
```
Comment (MarcoGorelli) [inline doc/source/user_guide/groupby.rst:208]: why make this change?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for your pr

File: pandas/core/shared_docs.py (hunk 1 above)
Comment (raanasn) [reply pandas/core/shared_docs.py]: Thanks for the adjustment.

File: doc/source/user_guide/groupby.rst (hunk 2 above)
Comment (raanasn) [reply doc/source/user_guide/groupby.rst:208]: In the comments of the [issue](https://github.com/pandas-dev/pandas/issues/47529) it was mentioned to add also a description in the "User Guide".

Comment (rhshadrach) [reply doc/source/user_guide/groupby.rst:208]: @raanasn - it looks like you're adding an unnecessary space before the period here.

Comment (MarcoGorelli) [reply doc/source/user_guide/groupby.rst:208]: exactly - if you revert this the rest look good to me

Comment (raanasn) [reply doc/source/user_guide/groupby.rst]: Typo, thanks!!

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @raanasn!

**Pull request merged**